app/controllers/fhir/r4/api_controller.rb

```diff
@@ -127,6 +127,8 @@ def update
     end
 
     status_ok(resource.as_fhir) && return
+  rescue JSON::ParserError
+    status_bad_request(['Failed to parse JSON'])
```

Comment:
I think this message kind of sounds like it was a server error (even though the status code says otherwise). I would just reword this to "Invalid JSON" or "Invalid JSON in request body" so it's super clear to the client that the issue was with their request.

Reply:
Good point, updated the error in https://github.com/SaraAlert/SaraAlert/pull/541/commits/c3d82053c36503adfdb5b2c38de6ff8e24b24c67.